examples/cmd/main.go

```diff
@@ -143,39 +209,80 @@ func (c *DownloaderCmd) ExecuteBody(logger log.Logger, fn bodyFunc) error {
 		if err = database.Migrate(c.DB); err != nil && err != migrate.ErrNoChange {
 			return err
 		}
+	}
 
-		downloader, err = github.NewDownloader(client, db)
+	downloadersPool, err := c.buildDownloadersPool(logger, db)
+	if err != nil {
+		return err
 	}
 
-	rate0, err := downloader.RateRemaining(context.TODO())
+	err = downloadersPool.Begin()
 	if err != nil {
 		return err
 	}
-	t0 := time.Now()
 
-	err = fn(client, downloader)
+	err = fn(logger, downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	err = downloader.SetCurrent(c.Version)
+	err = c.commit(downloadersPool)
 	if err != nil {
 		return err
 	}
 
-	if c.Cleanup {
-		return downloader.Cleanup(c.Version)
+	stats, err := downloadersPool.End()
+
+	logger = logger.With(log.Fields{"total-elapsed": stats.Elapsed})
```

Comment:
this logging is confusing.
It says each token was used `stats.Elapsed` which is not true as you can see in your logs. Most of the time it was downloading `chi` repo. And as we use only 1 token per repo `total-elapsed` can't be the same for both tokens.
I would propose to log it separately without relation to the tokens.

Reply:
you're right, it's not actually related to the tokens, thanks!